pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
@dave-tucker I believe that is because of startup order. I have a getty set as init here. Look at the output:

```
(ns: getty) linuxkit-025000000009:~# ls /etc/init.d
000-issue       001-getty       010-onboot      020-containerd
```

getty will be run before any of the onboot and before containerd. So you don't see it.

At least the init is easily fixable. The `init.d` file for getty is:

```
INSECURE=true /usr/bin/rungetty.sh
```

We could easily set it to 

```
INSECURE=true INITGETTY=true /usr/bin/rungetty.sh
```

and then check for `INITGETTY` in `rungetty.sh`. Again, we could get hit by someone setting the env var on their own, but we can document it.